app/domain/authentication/authn_jwt/restriction_validation/validate_restriction_name.rb

```diff
@@ -0,0 +1,35 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module RestrictionValidation
+      # Raises error if name of restriction is not according to the nested format
+      #
+      # Valid restriction example:
+      #   - authn-jwt/gitlab/a
+      #   - authn-jwt/gitlab/a/b
+      #   - authn-jwt/gitlab/a/b/c
+      #
+      # Invalid restriction example:
+      #   - authn-jwt/gitlab/a[4]/b
+      #   - authn-jwt/gitlab/a//b
+      #   - authn-jwt/gitlab/a/b[2]
+      #   - authn-jwt/gitlab/a/b/c
+      ValidateRestrictionName = CommandClass.new(
+        dependencies: {},
+        inputs: %i[restriction]
+      ) do
+        def call
+          validate_annotation_name
+        end
+
+        private
+
+        def validate_annotation_name
+          restriction_name = @restriction.name
+          raise Errors::Authentication::AuthnJwt::InvalidClaimPath.new(restriction_name, PURE_NESTED_CLAIM_NAME_REGEX) if restriction_name.nil? || !restriction_name.match?(PURE_NESTED_CLAIM_NAME_REGEX)
```

Comment:
if the line is too long then you shouldn't use a guard
```suggestion
          restriction_name = @restriction.name
          if restriction_name.nil? || !restriction_name.match?(PURE_NESTED_CLAIM_NAME_REGEX)
            raise Errors::Authentication::AuthnJwt::InvalidClaimPath.new(
              restriction_name, 
              PURE_NESTED_CLAIM_NAME_REGEX
            ) 
          end
          
```